### PR TITLE
Do not include error details in HTTP response

### PR DESCRIPTION
**Pull request by raboof**
Inspired by #2018

### CODE REVIEW

**Comment by raboof** [comment]
done

File: docs/src/main/paradox/routing-dsl/exception-handling.md [hunk 1]
```diff
@@ -94,6 +94,19 @@ please refer to @ref[the section above](exception-handling.md#exception-handling
 if the entity is not properly rejected or cancelled on the client side.
 @@@
 
+## Including sensitive data in exceptions
```
Comment (jrudolph) [inline docs/src/main/paradox/routing-dsl/exception-handling.md]: :+1: for adding docs.

**Comment by kag0** [comment]
@raboof this supersedes https://github.com/akka/akka-http/pull/2018 but imo we still need to make this change https://github.com/akka/akka-http/pull/2018/commits/a96e3fbc7cc10d58a2071050a314a0aafbc03cb, and the `formatPretty` [here](https://github.com/akka/akka-http/pull/2314/files#diff-a3c2fe9da784b62492e0c89750710e93R55) should probably be `format(settings.verboseErrorMessages)`.
The rationale being that we shouldn't log potentially sensitive data unless the user has configured it. I think logging sensitive data should be disabled by default, for example users who might get into trouble with compliance requirements not knowing that detail logging is enabled.

**Comment by raboof** [comment]
@kag0 thanks for pointing that out. That seems like a separate consideration, given that `settings.verboseErrorMessages` is explicitly about the client output, not about the logging (https://github.com/akka/akka-http/blob/c733ab41e0cb843c9506e96ad9c7a825c2aa7468/akka-http/src/main/resources/reference.conf#L10). Could you open a separate ticket/PR?

**Pull request merged**